scripts/20210517_validation-object/script.js

```diff
@@ -0,0 +1,283 @@
+      customVal: '$form_fields.ValidationOptions.customVal',
+      customMin: '$form_fields.ValidationOptions.customMin',
+      customMax: '$form_fields.ValidationOptions.customMax',
+    },
+  },
+  {
+    $match: {
+      $and: [
+        {
+          selectedValidation: 'Minimum',
+        },
+        {
+          $expr: { $ne: ['$customMin', '$customVal'] },
+        },
+      ],
+    },
+  },
+])
+
+// b. customVal === customMax when selectedValidation === 'Maximum'
+// Expect 0 records
+
+db.getCollection('forms').aggregate([
+  { $unwind: '$form_fields' },
+  {
+    $match: {
+      'form_fields.fieldType': { $in: ['textfield', 'textarea', 'number'] },
+    },
+  },
+  {
+    $project: {
+      form_fields: 1,
+      selectedValidation: '$form_fields.ValidationOptions.selectedValidation',
+      customVal: '$form_fields.ValidationOptions.customVal',
+      customMin: '$form_fields.ValidationOptions.customMin',
+      customMax: '$form_fields.ValidationOptions.customMax',
+    },
+  },
+  {
+    $match: {
+      $and: [
+        {
+          selectedValidation: 'Maximum',
+        },
+        {
+          $expr: { $ne: ['$customMax', '$customVal'] },
+        },
+      ],
+    },
+  },
+])
+
+// c. customVal === customMin === customMax when selectedValidation === 'Exact'
+// Expect 0 records
+
+db.getCollection('forms').aggregate([
+  { $unwind: '$form_fields' },
+  {
+    $match: {
+      'form_fields.fieldType': { $in: ['textfield', 'textarea', 'number'] },
+    },
+  },
+  {
+    $project: {
+      form_fields: 1,
+      selectedValidation: '$form_fields.ValidationOptions.selectedValidation',
+      customVal: '$form_fields.ValidationOptions.customVal',
+      customMin: '$form_fields.ValidationOptions.customMin',
+      customMax: '$form_fields.ValidationOptions.customMax',
+    },
+  },
+  {
+    $match: {
+      $and: [
+        {
+          selectedValidation: 'Exact',
+        },
+        {
+          $or: [
+            {
+              $expr: { $ne: ['$customMax', '$customVal'] },
+            },
+            {
+              $expr: { $ne: ['$customMin', '$customVal'] },
+            },
+          ],
+        },
+      ],
+    },
+  },
+])
+
+// d. selectedValidation === 'Range' should not exist
+// Expect 0 records
+
+db.getCollection('forms').aggregate([
+  { $unwind: '$form_fields' },
+  {
+    $match: {
+      'form_fields.fieldType': { $in: ['textfield', 'textarea', 'number'] },
+    },
+  },
+  {
+    $project: {
+      form_fields: 1,
+      selectedValidation: '$form_fields.ValidationOptions.selectedValidation',
+    },
+  },
+  {
+    $match: {
+      selectedValidation: 'Range',
+    },
+  },
+])
+
+/**
+ * Part 2: Unset form_fields.ValidationOptions.customMin and
+ * form_fields.ValidationOptions.customMax
+ * for all short text, long text and number fields.
+ */
+
+// ********** BEFORE **********
+
+// Count number of forms with (short text, long text and number fields) and
+// with ValidationOptions.customMin
```

Comment:
maybe we should do a count for `ValidationOptions.customMin` and `customMax` for all other field types as well, as a sanity check to make sure that we haven't affected this key in decimal and checkbox fields

Reply:
thanks! i've added in